snow/validators/retriever_test.go

```diff
@@ -0,0 +1,69 @@
+package validators
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/flare-foundation/flare/ids"
+	"github.com/flare-foundation/flare/utils/constants"
+)
+
+type testRetriever struct {
+	m map[ids.ID]Set
+}
+
+var Counter = 0
+var GetValidatorsCallCounter = 0
```

Comment:
Careful, this means that the `validators` package now exposes `Counter` and `GetValidatorsCallCounter` as global variables 😱 

I don't understand what the point of having those counters here is either, as well as having the logic for the retriever mock hardcoded here. If instead of being hardcoded here, it was set in the test (avoids having implicit behavior defined outside of the test too) it would look something like this:

```go
t.Run("my test case", func(t *testing.T){
	t.Parallel()

	callsCount := 0
	validators := make(map[ids.ID]Set)
	retrieverMock := &RetrieverMock{
		GetValidatorsFunc: func(blockID ids.ID) (Set, error) {
			callsCount++
			// ... Rest of the logic here. Use the `validators` map defined in this scope to keep the mock completely
			// basic and without any logic besides forwarding function calls.
		}
	}

	assert.Equal(t, 12, callsCount)
	// etc.
})
```

Reply:
This commit is an attempt in the direction mentioned above(although doesnt include all cases): https://github.com/flare-foundation/flare/commit/455ae6f3939407f64864777c06c6fa1dfa34cfc5

Please let me know if this somewhat aligns with what you meant or if I should make any other changes. 

Can you give any suggestion on how to test error case properly as currently I am simply forcing an error?